src/components/manage/Widgets/SiteLogoWidget.jsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * SiteLogoWidget component.
+ * @module components/manage/Widgets/SiteLogoWidget
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Image, Dimmer } from 'semantic-ui-react';
+import { readAsDataURL } from 'promise-file-reader';
+import { injectIntl } from 'react-intl';
+import deleteSVG from '@plone/volto/icons/delete.svg';
+import { Icon, FormFieldWrapper } from '@plone/volto/components';
+import loadable from '@loadable/component';
+import { defineMessages, useIntl } from 'react-intl';
+import { toPublicURL } from '@plone/volto/helpers';
+
+const imageMimetypes = [
+  'image/png',
+  'image/jpeg',
+  'image/webp',
+  'image/jpg',
+  'image/gif',
+  'image/svg+xml',
+];
+const Dropzone = loadable(() => import('react-dropzone'));
+
+const messages = defineMessages({
+  releaseDrag: {
+    id: 'Drop files here ...',
+    defaultMessage: 'Drop files here ...',
+  },
+  editFile: {
+    id: 'Drop file here to replace the existing file',
+    defaultMessage: 'Drop file here to replace the existing file',
+  },
+  fileDrag: {
+    id: 'Drop file here to upload a new file',
+    defaultMessage: 'Drop file here to upload a new file',
+  },
+  replaceFile: {
+    id: 'Replace existing file',
+    defaultMessage: 'Replace existing file',
+  },
+  addNewFile: {
+    id: 'Choose a file',
+    defaultMessage: 'Choose a file',
+  },
+});
+
+/**
+ * SiteLogoWidget component class.
+ * @function SiteLogoWidget
+ * @returns {string} Markup of the component.
+ *
+ * To use it, in schema properties, declare a field like:
+ *
+ * ```jsx
+ * {
+ *  title: "File",
+ *  widget: 'file',
+ * }
+ * ```
+ * or:
+ *
+ * ```jsx
+ * {
+ *  title: "File",
+ *  type: 'object',
+ * }
+ * ```
+ *
+ */
+const SiteLogoWidget = (props) => {
+  const { id, value, onChange, isDisabled } = props;
+  const intl = useIntl();
+
+  const fileName = value?.split(';')[0];
+  const imgsrc = fileName
+    ? `${toPublicURL('/')}@@site-logo/${atob(
+        fileName.replace('filenameb64:', ''),
+      )}`
+    : '';
+
+  /**
+   * Drop handler
+   * @method onDrop
+   * @param {array} files File objects
+   * @returns {undefined}
+   */
+  const onDrop = (files) => {
+    const file = files[0];
```

Comment:
This should be updated to use the validateFileUploadSize helper I just added in https://github.com/plone/volto/pull/4868

Reply:
Added the validation, thanks!